pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -837,6 +837,15 @@ func (r *ReconcileClusterDeployment) startNewProvision(
 		}
 	}
 
+	if err := r.setupAWSCredentialForAssumeRole(cd); err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			// Couldn't create the assume role credential secret for a reason other than it already exists.
+			// If the secret already exists, then we should just use that secret.
```

Comment:
Do we not need to worry about Secret contents potentially needing to change here?

Reply:
the secret generated wouldn't change as the inputs are fixed for multiple attempts.